### ISSUE TITLE
NancyContext.CurrentUser returns null when making http POST request [Demo.Authentication.Stateless.Website]

### ISSUE THREAD

**Issue by Adesun** [NancyFx/Nancy#1745]
most of  the code just copy the  project of Nancy.Demo.Authentication.Stateless.Website

the only difference is  changing GET into POST , then the code below throws exception when making http POST request.

```
namespace Nancy.Demo.Authentication.Stateless
{
    using Nancy.Demo.Authentication.Stateless.Models;
    using Nancy.Security;

    public class SecureModule : NancyModule
    {
        //by this time, the api key should have already been pulled out of our querystring
        //and, using the api key, an identity assigned to our NancyContext
        public SecureModule()
        {
            this.RequiresAuthentication();

           Post["secure"] = x =>
            {
                // exception throw
                // Nancy.RequestExecutionException: Oh noes! ---> System.NullReferenceException:
                var identity = (DemoUserIdentity)this.Context.CurrentUser;

                //return the secure information in a json response
                var userModel = new UserModel(identity.UserName);
                return this.Response.AsJson(new
                {
                    SecureContent = "here's some secure content that you can only see if you provide a correct apiKey",
                    User = identity.UserName
                });
            };
        }
    }
}
```

**Comment by feeeper** [comment]
If you go to the `StatelessAuthBootstrapper.cs` you can see that code:
``` C#
var configuration =
  new StatelessAuthenticationConfiguration(nancyContext =>
  {
    //for now, we will pull the apiKey from the querystring,
    //but you can pull it from any part of the NancyContext
    var apiKey = (string) nancyContext.Request.Query.ApiKey.Value;

    //get the user identity however you choose to (for now, using a static class/method)
    return UserDatabase.GetUserFromApiKey(apiKey);
  });
```
So you just need to pass your `apiKey` through query string like that:
``` JavaScript
$.ajax({
  type: 'POST', 
  url: 'restapi/secure?ApiKey=' + ApiToken.Get().Key, 
  success: function(s) { 
    console.log('success'); 
    console.log(s); 
  }, 
  dataType: "json" 
});
```

**Comment by jchannon** [comment]
If either of you want to send a PR for that area it would be greatly appreciated!

**Comment by khellang** [comment]
Closed by #2312
